### ISSUE TITLE
Allow copying from a known Docker image

### ISSUE THREAD

**Issue by MetalArend** [hadolint/hadolint#859]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Allow copying from a known Docker image.

### Actual behavior

Giving error "DL3022 warning: `COPY --from` should reference a previously defined `FROM` alias"

### Steps to reproduce the behavior

```Dockerfile
FROM scratch
COPY --from=hello-world /hello /hello
```

`docker run --rm -i docker.io/hadolint/hadolint < Dockerfile`

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile** (if relevant)

```Dockerfile
FROM scratch
COPY --from=hello-world /hello /hello
```

**Additional environment details (OS, stack version, etc.)**

**Comment by akostadinov** [comment]
Duplicate of #718
Seems to be a regression from #815

**Comment by Wirone** [comment]
@akostadinov it's not a regression, my changes support only `image:tag` format, because without `:` it's impossible to distinguish previous layers' names from images' names. @MetalArend If you use Hadolint and want to keep `DL3022` rule, you should do `COPY --from=hello-world:latest` (explicit `latest`) 🙂.

**Comment by akostadinov** [comment]
@Wirone , I use a tag. I have:
```
...
ARG PORTA_IMAGE=quay.io/3scale/porta:nightly
COPY --from=$PORTA_IMAGE /opt/system/config/standalone.sphinx.conf "/etc/sphinx/system.sphinx.conf"
...
```

IMO this is a bug, maybe not a regression.

Also honestly I think the error is not ideal. Linter for me is more about recommended style. Not to catch errors especially in cases where syntax is ok, just we can't figure out whether value is correct or not. When one tries to build an image, if things are simply wrong, the build will fail. But if things are bad but not wrong, then linter is really useful.

This check would be more useful if it checks whether such stage exists -> pass, if it doesn't exist but there is a tag -> also pass. And if there is no such stage and no tag -> warn: use tag when referring to images. Same as DL3006 I guess.

 Anyway, I disabled this check so I'm fine. Just tried to be helpful for the project.

**Comment by Wirone** [comment]
@akostadinov You use tag, but also you use `ARG` so `--from` does not contain actual value. Hadolint is a static analyser, AFAIK it can't resolve your variable and run check against actual value which is stored in `$PORTA_IMAGE`. Also, it does not make sense to keep image in the `ARG` because passing any other image there would cause the error (non-existing `/opt/system/config/standalone.sphinx.conf`). It would be better to do this:

```dockerfile
ARG PORTA_VERSION=nightly
COPY --from=quay.io/3scale/porta:$PORTA_VERSION /opt/system/config/standalone.sphinx.conf "/etc/sphinx/system.sphinx.conf"
```

and then Hadolint should work even with the variable (the key part is `:`).

**Comment by akostadinov** [comment]
I guess this is offtopic, but we have productized and community images. Actually the bigger issue is that within the build system it uses some internal container repositories. So not practical to hardcode the repo. But in general your suggestion is sensible.

**Comment by Wirone** [comment]
So you can use

```dockerfile
ARG PORTA_IMAGE=quay.io/3scale/porta
ARG PORTA_VERSION=nightly
COPY --from=$PORTA_IMAGE:$PORTA_VERSION /opt/system/config/standalone.sphinx.conf "/etc/sphinx/system.sphinx.conf"
```
and keep full flexibility of use internal repositories, but with `image:tag` format that Hadolint would understand.

**Comment by bdovaz** [comment]
I just encountered this problem but in my case using multiple contexts:

https://www.docker.com/blog/dockerfiles-now-support-multiple-build-contexts/
